### PR TITLE
add sliver cross axis constrained

### LINKED ISSUE #1: Add sliver that limits crossAxisExtent
Should have these constructor properties:
- maxCrossAxisExtent
- alignment (perhaps just a double between -1 and +1

### PR DESCRIPTION
**Pull request by remonh87**
Added basic SliverCrossAxisConstrained widget that limits the cross axis and centers the child sliver. Quite useful for tablets

Fixes #1

### CODE REVIEW

**Comment by remonh87** [comment]
Fixes #1

File: lib/src/sliver_cross_axis_constrained.dart [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+
+    final childLayoutGeometry = child.geometry;
+    if (childLayoutGeometry.scrollOffsetCorrection != null) {
+      geometry = SliverGeometry(
+        scrollOffsetCorrection: childLayoutGeometry.scrollOffsetCorrection,
+      );
+      return;
+    }
+
+    geometry = SliverGeometry(
```
Comment (Kavantix) [inline lib/src/sliver_cross_axis_constrained.dart]: We probably should just return the child's geometry

File: lib/src/sliver_cross_axis_constrained.dart [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+      child.parentData = SliverPhysicalParentData();
+  }
+
+  @override
+  void performLayout() {
+    child.layout(
+        constraints.copyWith(
+            crossAxisExtent:
+                min(constraints.crossAxisExtent, _maxCrossAxisExtent)),
+        parentUsesSize: true);
```
Comment (Kavantix) [inline lib/src/sliver_cross_axis_constrained.dart]: Nit: I like adding trailing commas on all non one liners

File: lib/src/sliver_cross_axis_constrained.dart [hunk 3]
```diff
@@ -0,0 +1,176 @@
+import 'dart:math';
+
+import 'package:flutter/rendering.dart';
+import 'package:flutter/widgets.dart';
+
+/// Constraints and centers the [child] sliver to a maximum cross axis extent
+///  specified by [maxCrossAxisExtent].
+class SliverCrossAxisConstrained extends SingleChildRenderObjectWidget {
+  const SliverCrossAxisConstrained({
+    @required this.maxCrossAxisExtent,
```
Comment (Kavantix) [inline lib/src/sliver_cross_axis_constrained.dart:10]: Future work: Add ability to use a padding value

**Pull request merged**